### PR TITLE
feat(docs): show inflight methods next to preflight

### LINKED ISSUE #3165: Resources should show both preflight methods and inflight methods
notice that [bucket api docs](https://www.winglang.io/docs/standard-library/cloud/api-reference#bucket-) only shows the preflight api, and you have to navigate to [IBucketClient api doc](https://www.winglang.io/docs/standard-library/cloud/api-reference#@winglang/sdk.cloud.IBucketClient) to see the inflight docs

### PR DESCRIPTION
**Pull request by tsuf239**
fixes #3165 
* removes "inflight client" line in the resource documentation
* moving the inflight methods under the preflight methods section, and removing the inflight client's protocol
* changing the name of the methods sections related to the existence of inflight methods
* UPDATE: adding inflight client annotation to the api resource (it was missing)
https://www.loom.com/share/7d2b166226044fdeacfe0da9a922a3e8?sid=25263ac4-77df-4951-aec4-8f59d0aa8914

## Checklist

- [x] Title matches [Winglang's style guide](https://www.winglang.io/contributing/start-here/pull_requests#how-are-pull-request-titles-formatted)
- [x] Description explains motivation and solution
- [ ] Tests added (always)
- [ ] Docs updated (only required for features)
- [ ] Added `pr/e2e-full` label if this feature requires end-to-end testing

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Monada Contribution License](https://www.winglang.io/terms-and-policies/contribution-license.html)*.

### CODE REVIEW

**Comment by tsuf239** [comment]
image

File: apps/jsii-docgen/src/docgen/view/api-reference.ts [hunk 1]
```diff
@@ -63,10 +64,28 @@ export class ApiReference {
...
+    classes.forEach((c) => {
+      const inflight = getInflight(c.docs);
+      if (inflight) {
+        c.inflightId = MarkdownDocument.sanitize(inflight);
+      }
+    });
+
+    const inflightIds: string[] = classes
+      .map(({ inflightId }) => inflightId)
+      .filter((id) => typeof id === "string") as string[];
```
Comment (Chriscbr) [inline apps/jsii-docgen/src/docgen/view/api-reference.ts:76]: Are these referring to the type names like "@winglang/sdk.cloud.IBucketClient"? If so the name "fqn" is more typical (it stands for Fully Qualified Name and is used in a few other places)
```suggestion
    const inflightFqns: string[] = classes
      .map(({ inflightFqn }) => inflightFqn)
      .filter((fqn) => typeof fqn === "string") as string[];
```

File: docs/docs/04-standard-library/01-cloud/queue.md [hunk 2]
```diff
@@ -108,12 +106,21 @@ new cloud.Queue(props?: QueueProps);
...
+| **Name** | **Description** |
+| --- | --- |
+| <code><a href="#@winglang/sdk.cloud.IQueueClient.approxSize">approxSize</a></code> | Retrieve the approximate number of messages in the queue. |
+| <code><a href="#@winglang/sdk.cloud.IQueueClient.pop">pop</a></code> | Pop a message from the queue. |
+| <code><a href="#@winglang/sdk.cloud.IQueueClient.purge">purge</a></code> | Purge all of the messages in the queue. |
+| <code><a href="#@winglang/sdk.cloud.IQueueClient.push">push</a></code> | Push a message to the queue. |
+
 ---
 
 ##### `setConsumer` <a name="setConsumer" id="@winglang/sdk.cloud.Queue.setConsumer"></a>
```
Comment (Chriscbr) [inline docs/docs/04-standard-library/01-cloud/queue.md:126]: The ordering of headings here might be confusing:
```
- Preflight Methods
- Inflight Methods
  - setConsumer
  - ...
  - approxSize
  - ...
```

Would it make sense to put a single title named "Methods" and have the two tables under it?

**Review by Chriscbr — APPROVED** [review]
Awesome!

File: docs/docs/04-standard-library/01-cloud/queue.md (hunk 2 above)
Comment (Chriscbr) [reply docs/docs/04-standard-library/01-cloud/queue.md:126]: Another option could be to put the `preflight` or `inflight` keyword next to the method names in the table. I'm also okay with the current solution

Comment (tsuf239) [reply docs/docs/04-standard-library/01-cloud/queue.md:126]: Great idea! Eyal and I liked that :) 
image

**Pull request merged**